qiskit_machine_learning/algorithms/trainable_model.py

```diff
@@ -202,7 +205,8 @@ def _choose_initial_point(self) -> np.ndarray:
         if self._warm_start and self._fit_result is not None:
             self._initial_point = self._fit_result[0]
         elif self._initial_point is None:
-            self._initial_point = np.random.rand(self._neural_network.num_weights)
+            rand_state = RandomState(MT19937(SeedSequence(algorithm_globals.random_seed)))
+            self._initial_point = rand_state.rand(self._neural_network.num_weights)
```

Comment:
I don't fully understand what is going on here, but the reason of the problem is not here. The test `test_neural_network_classifier.test_classifier_with_circuit_qnn_and_cross_entropy` fails sometimes despite `initial_point` initialized as `np.array([0.5] * ansatz.num_parameters)`. So, we start optimization from the same point every time, but sometimes we end up having a score of `0.4`.

Reply:
If I understand it correctly, @woodsp-ibm asked for utilization of `algorithm_globals.random_seed` in `VQC`in https://github.com/Qiskit/qiskit-machine-learning/issues/158#issue-942421425 .

Also, I've passed `algorithm_globals.random_seed` in `QSVC` constructor as `random_state` parameter, so that it "reacts" to the globally-set seed.

Reply:
@adekusar-drl Maybe it'd be better to make a separate issue for the problem with `test_neural_network_classifier.test_classifier_with_circuit_qnn_and_cross_entropy`? It'd prevent confusion, as this one seems to be specifically about `QSVC` and `VQC`.

Reply:
The problem is not only in `VQC`. For instance, `test_neural_network_regressor.test_regressor_with_opflow_qnn()` fails for the same reason sometimes. `VQC`, in turns, extends `NeuralNetworkClassifier` and if the latter fails, the former fails as well.
If we are talking about algorithm globals as stated in the issue, then `np.random.rand(self._neural_network.num_weights)` is to be replaced with `algorithm_globals.random.random(...)`. Fine with me, we can just fix this, but such a fix won't fix the problem we try to cope.

Reply:
@adekusar-drl  The issue I had raised #158 was a) because I noticed direct use of np.random rather than the algorithm globals that is used elsewhere that we can seed centrally for reproducibility etc.  And b) for qsvc it was that sklearn svc has a random aspect that can be seeded, it had been noted to me that the results varied despite setting algorithm globals, the simulator seeds etc. And while in our subclass you can certainly still set the sklearn random state via kwargs, one needs to look at sklearn to figure this out, so simply setting this via algo globals random seed in the qsvc I figured would make it easier for users. So I raised the issue just with these points in mind, it was not motivated at all by the random unit test failure, that occurs.

I must admit I had imagine the change of

> self._initial_point = np.random.rand(self._neural_network.num_weights)

to algorithm_globals to be simply

> self._initial_point = algorithm_globals.random.rand(self._neural_network.num_weights)

where instead of using np global random object it used the algorithms global one.

Reply:
@woodsp-ibm Thank you for the explanation! I've rewritten the random number generation according to your suggestion.

Reply:
@woodsp-ibm OK, clear. I was expecting slightly more in this PR :)